src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRazorPageFactoryProvider.cs

```diff
@@ -62,30 +62,12 @@ public RazorPageFactoryResult CreateFactory(string relativePath)
             var result = CompilerCache.GetOrAdd(relativePath, _compileDelegate);
             if (result.Success)
             {
-                var pageFactory = GetPageFactory(result.CompilationResult.CompiledType, relativePath);
-                return new RazorPageFactoryResult(pageFactory, result.ExpirationTokens);
+                return new RazorPageFactoryResult(result.PageFactory, result.ExpirationTokens);
             }
             else
             {
                 return new RazorPageFactoryResult(result.ExpirationTokens);
             }
         }
-
-        /// <summary>
-        /// Creates a factory for <see cref="IRazorPage"/>.
-        /// </summary>
-        /// <param name="compiledType">The <see cref="Type"/> to produce an instance of <see cref="IRazorPage"/>
-        /// from.</param>
-        /// <param name="relativePath">The application relative path of the page.</param>
-        /// <returns>A factory for <paramref name="compiledType"/>.</returns>
-        protected virtual Func<IRazorPage> GetPageFactory(Type compiledType, string relativePath)
```

Comment:
This was meant to be analogous to ControllerActivator but we don't give the user a type now to do anything useful. I'm ok with removing this. cc @Eilon for API change